### PR TITLE
Remove double encoding in MathML via Unicode MathML spaces

### LINKED ISSUE #1782: katex.render double encodes HTML entities in MathML
`katex.render` double encodes HTML entities in MathML. For example, the following formula:

```latex
x < y
```

Produces an element with the following MathML:

```html
<math>
  <semantics>
    <mrow><mi>x</mi><mo>&amp;lt;</mo><msup><mi>y</mi><mn>2</mn></msup></mrow>
    <annotation encoding=\"application/x-tex\">x &amp;lt; y^2</annotation>
  </semantics>
</math>
```

`&amp;lt;` is produced, `&lt;` is expected.

`katex.renderToString` is not affected by the issue (`element.innerHTML = katex.renderToString` produces expected output).

### PR DESCRIPTION
**Pull request by edemaine**
We used to have a complex mechanic for escaping strings, marking them as `needsEscape = false`, and then not escaping those strings (for combining strings in `\operatorname`).  But this doesn't really work with `render`ing directly to a node, as `document.createTextNode` can't be stopped from escaping.

I've thus removed this mechanic, which required the following changes:

* Switch MathML "smart space" encoding to use Unicode instead of `&LongNames;` (which weren't working with `render` for the same reason).
* ~~Hack our HTML/MathML serializer to not use `String.trim`, which wrecks havoc with emitted Unicode spaces. (See also https://github.com/rayrutjes/diffable-html/issues/14)~~ Now fixed thanks to https://github.com/rayrutjes/diffable-html/pull/15

Now `toText()` doesn't escape, so strings concatenate in unescaped form, and `toHTML()` only does the necessary escaping.  Thus fix #1782.

This PR replaces #1793.

### CODE REVIEW

File: src/mathMLTree.js [hunk 1]
```diff
@@ -126,34 +126,32 @@ export class MathNode implements MathDomNode {
...
      * Converts the text node into escaped HTML markup
      * (representing the text itself).
      */
     toMarkup(): string {
-        return this.toText();
+        return utils.escape(this.toText());
     }
 
     /**
      * Converts the text node into an escaped string
```
Comment (ylemkimon) [inline src/mathMLTree.js]:
```suggestion
     * Converts the text node into a string
```

File: src/utils.js [hunk 2]
```diff
@@ -95,6 +95,7 @@ export default {
     contains,
     deflt,
     escape,
+    unescape,
```
Comment (ylemkimon) [inline src/utils.js]:
```suggestion
```

File: test/setup.js [hunk 3]
```diff
@@ -8,7 +9,29 @@ import {
...
+// generating extra spaces.
+expect.addSnapshotSerializer({
+    print(val) {
+        const trim = String.prototype.trim;
+        String.prototype.trim = function() { return this; };
+        try {
+            return toDiffableHtml(val);
+        } finally {
+            String.prototype.trim = trim;
+        }
```
Comment (ylemkimon) [inline test/setup.js]: Is there any other library that can do this?

Comment (edemaine) [reply test/setup.js]: We could search.  But also a new version of this library should now do what we want, thanks to https://github.com/rayrutjes/diffable-html/pull/15

**Pull request merged**